src/main/java/uk/gov/hmcts/divorce/caseworker/service/ReIssueApplicationService.java

```diff
@@ -87,31 +91,56 @@ private CaseDetails<CaseData, State> updateCase(CaseDetails<CaseData, State> cas
             return caseTasks(
                 setPostIssueState,
                 setReIssueAndDueDate,
+                setNoticeOfProceedingDetailsForRespondent,
                 generateApplicant1NoticeOfProceeding,
                 generateApplicant2NoticeOfProceedings,
                 generateDivorceApplication,
-                generateD10Form,
-                sendAosPackToRespondent,
-                sendAosPackToApplicant,
-                sendApplicationIssueNotifications
+                generateD10Form
             ).run(caseDetails);
         } else if (REISSUE_CASE.equals(reissueOption)) {
             log.info("For case id {} processing complete reissue ", caseDetails.getId());
             return caseTasks(
                 setPostIssueState,
                 setReIssueAndDueDate,
+                setNoticeOfProceedingDetailsForRespondent,
                 generateApplicant1NoticeOfProceeding,
                 generateApplicant2NoticeOfProceedings,
                 generateDivorceApplication,
-                generateD10Form,
+                generateD10Form
+            ).run(caseDetails);
+        } else {
+            log.info("For case id {} invalid reissue option hence not processing reissue application ", caseDetails.getId());
+            throw new ReissueProcessingException(
+                "Exception occurred while processing reissue application for case id " + caseDetails.getId()
+            );
+        }
+    }
+
+    public void sendNotifications(CaseDetails<CaseData, State> caseDetails, ReissueOption reissueOption) {
+        if (DIGITAL_AOS.equals(reissueOption)) {
+            log.info("For case id {} sending reissue notifications for digital aos ", caseDetails.getId());
+            caseTasks(
+                sendApplicationIssueNotifications
+            ).run(caseDetails);
+        } else if (OFFLINE_AOS.equals(reissueOption)) {
+            log.info("For case id {} sending reissue notifications for offline aos ", caseDetails.getId());
+
+            caseTasks(
+                sendAosPackToRespondent,
+                sendAosPackToApplicant,
+                sendApplicationIssueNotifications
```

Comment:
Only question I have is if we should send email notifications to offline_aos applicants? Approving as rest looks fine and I see it was sending them before the refactor